packages/superset-ui-time-format/src/factories/getTimeFormatterForGranularity.ts

```diff
@@ -39,16 +41,30 @@ const formats = {
   P1M: '%Y-%m', // month
   'P0.25Y': '%Y Q%q', // quarter
   P1Y: '%Y', // year
-  // d3-time-format weeks does not support weeks start on Sunday
   '1969-12-28T00:00:00Z/P1W': SUNDAY_BASED_WEEK, // 'week_start_sunday'
   '1969-12-29T00:00:00Z/P1W': MONDAY_BASED_WEEK, // 'week_start_monday'
   'P1W/1970-01-03T00:00:00Z': SUNDAY_BASED_WEEK, // 'week_ending_saturday'
   'P1W/1970-01-04T00:00:00Z': MONDAY_BASED_WEEK, // 'week_ending_sunday'
 };
 
-export type TimeGranularity = keyof typeof formats;
+export type TimeGranularity =
```

Comment:
There is a known issue with re-exporting type for library. 

```
export {
  default as getTimeFormatterForGranularity,
  TimeGranularity,
} from './factories/getTimeFormatterForGranularity';
```
there will be issue with `TimeGranularity` in the consumer app.

Have to create `./types.ts` and put this type there 
then in index 

```ts
export * from './types.ts'
```

Reply:
Does it mean the consumer apps will have to `import from '@sueprset-ui/time-format/src/types` too?

Reply:
no they can 

`import { TimeGranularity } from '@superset-ui/time-format';`